scene/animation/animation_node_state_machine.cpp

```diff
@@ -1322,7 +1322,8 @@ bool AnimationNodeStateMachine::are_ends_reset() const {
 
 bool AnimationNodeStateMachine::can_edit_node(const StringName &p_name) const {
 	if (states.has(p_name)) {
-		return !(states[p_name].node->is_class("AnimationNodeStartState") || states[p_name].node->is_class("AnimationNodeEndState"));
+		const Ref<AnimationNode> &anode = states[p_name].node;
+		return !(Ref<AnimationNodeStartState>(anode).is_valid() || Ref<AnimationNodeEndState>(anode).is_valid());
```

Comment:
Note that I didn't actually check whether this needed to change to work. But in general relying on casting pointers seems better than a string check through ClassDB.